dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/mail/MailService.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package edu.harvard.iq.dataverse.mail;
+
+import edu.harvard.iq.dataverse.DataverseServiceBean;
+import edu.harvard.iq.dataverse.notification.dto.EmailNotificationDto;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean.Key;
+import io.vavr.Tuple2;
+import io.vavr.control.Try;
+import org.apache.commons.lang.StringUtils;
+import org.simplejavamail.email.Email;
+import org.simplejavamail.email.EmailBuilder;
+import org.simplejavamail.mailer.Mailer;
+import org.simplejavamail.mailer.MailerBuilder;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Resource;
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+/**
+ * Service responsible for mail sending.
+ */
+@Stateless
+public class MailService implements java.io.Serializable {
+
+    private DataverseServiceBean dataverseService;
+    private SettingsServiceBean settingsService;
+    private MailMessageCreator mailMessageCreator;
+
+    private Mailer mailSender;
+
+    private static final Logger logger = Logger.getLogger(MailService.class.getCanonicalName());
+
+
+    @Resource(name = "mail/notifyMailSession")
+    private Session session;
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    @Deprecated /* JEE requirement */
+    MailService() {
+    }
+
+    @Inject
+    public MailService(DataverseServiceBean dataverseService, SettingsServiceBean settingsService, MailMessageCreator mailMessageCreator) {
+        this.dataverseService = dataverseService;
+        this.settingsService = settingsService;
+        this.mailMessageCreator = mailMessageCreator;
+    }
+
+    @PostConstruct
```

Comment:
Why in @PostConstruct and not in consturctor itself?

Reply:
It is a problem with dependency injection, since Mailer is library i can't inject it. So in unit tests it won't work, since it will throw an exception that session is null.